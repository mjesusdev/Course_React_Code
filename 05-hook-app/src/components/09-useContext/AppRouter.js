import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect,
  } from 'react-router-dom';

import { NavBar } from './NavBar';

import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                
                <NavBar />

                <div className="container">
                    <Switch>
                        <Route exact path="/" component={ HomeScreen } />

                        <Route exact path="/about" component={ AboutScreen } />
                        <Route exact path="/login" component={ LoginScreen } />
                        
                        <Redirect to="/" />

                    </Switch>
                </div>
            </div>
        </Router>
    )
}
