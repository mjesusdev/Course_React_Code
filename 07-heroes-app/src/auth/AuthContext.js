import { createContext } from 'react';

export const authContext = createContext();