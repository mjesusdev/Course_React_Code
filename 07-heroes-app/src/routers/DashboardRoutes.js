import React from 'react';
import { Routes, Route } from 'react-router-dom';

import { Navbar } from '../components/ui/Navbar';

import { DcScreen } from '../components/dc/DcScreen';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { SearchScreen } from '../components/search/SearchScreen';
import { HeroScreen } from '../components/heroes/HeroScreen';

export const DashboardRoutes = () => {
    return (
        <>
            <Navbar />

            <div className="container mt-2">
                <Routes>
                    <Route path="marvel" element={<MarvelScreen />} />
                    <Route path="dc" element={<DcScreen />} />

                    <Route path="search" element={<SearchScreen />} />
                    <Route path="hero/:heroeId" element={<HeroScreen />} />

                    <Route path="/" element={<MarvelScreen />}/>
                </Routes>
            </div>
        </>
    )
}
