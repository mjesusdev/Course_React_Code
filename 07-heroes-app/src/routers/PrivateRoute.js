import React from 'react';
import PropTypes from 'prop-types';

import { Route } from 'react-router';

export const PrivateRoute = ({
    isAutenticated,
    component: Component,
    ...rest
}) => {

    localStorage.setItem('lastPath', rest.location.pathname);

    return (
        <Route { ...rest } 
            component={ (props) => (
                ( isAutenticated )
                    ? ( <Component { ...props } /> )
                    : ( <Route path="login" render={() => <Redirect to="login" /> } />)
            )}
        />
    )
}

PrivateRoute.propTypes = {
    isAutenticated: PropTypes.bool.isRequired,
    component: PropTypes.func.isRequired
}