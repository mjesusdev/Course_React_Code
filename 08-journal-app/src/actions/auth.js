import { types } from "../types/types";
import { firebase, googleAuthProvider } from '../firebase/firebase-config';

// Sign in with Email & Password
export const startLoginEmailPassword = ( email, password ) => {
    return ( dispatch ) => {
        setTimeout(() => {
            dispatch( login(123, 'Peter') );
        }, 3000)
    }
}

// Register User with Email & Password
export const startRegisterUser = ( email, password, name ) => {
    return ( dispatch ) => {

        firebase.auth().createUserWithEmailAndPassword( email, password)
            .then( async ({ user }) => {

                await user.updateProfile({ displayName: name });

                dispatch(
                    login( user.uid, user.displayName )
                )
            })

            .catch( error => {
                console.log(error);
            })
    }
}

// Sign in with Google Account
export const startGoogleLogin = () => {
    return ( dispatch ) => {
        firebase.auth().signInWithPopup( googleAuthProvider )
            .then( ({ user }) => {
                dispatch(
                    login( user.uid, user.displayName )
                )
            });
    }
}

// Export
export const login = ( uid, displayName ) => ({
    type: types.login,
    payload: {
        uid,
        displayName
    }
})