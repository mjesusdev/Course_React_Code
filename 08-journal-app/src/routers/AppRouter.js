import React from 'react';
import {
    BrowserRouter as Router,
    Redirect,
    Route,
    Switch
} from 'react-router-dom';

import { JournalScreen } from '../components/journal/JournalScreen';
import { AuthRouter } from './AuthRouter';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <Switch> 
                    <Route  
                        path="/auth" 
                        component={ AuthRouter } 
                    />
                    
                    <Route 
                        exact
                        path="/"
                        component={ JournalScreen }
                    />

                    <Redirect to="/auth/login" />
                    
                </Switch>
            </div>
        </Router>
    )
}
