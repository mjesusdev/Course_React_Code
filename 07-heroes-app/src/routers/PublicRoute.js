import React, { useContext } from 'react';
import { Navigate } from 'react-router-dom';
import { authContext } from '../auth/authContext';

export const PublicRoute = ({ children }) => {

    const { user } = useContext(authContext);

    return user.logged
        ? <Navigate to="/marvel" />
        : children
}