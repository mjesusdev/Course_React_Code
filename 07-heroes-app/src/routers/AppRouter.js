import React, { useContext } from 'react'
import { Routes } from 'react-router-dom';

import { AuthContext } from '../auth/AuthContext';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';

export const AppRouter = () => {

    const { user } = useContext(AuthContext);

    return (
        <div>
            <Routes> 
                <PublicRoute
                    path="/login"
                    element={ <LoginScreen /> } 
                    isAutenticated={ user.logged }
                />
                
                <PrivateRoute 
                    path="/*"
                    element={ <DashboardRoutes /> }
                    isAutenticated={ user.logged }
                />
            </Routes>
        </div>
    )
}
