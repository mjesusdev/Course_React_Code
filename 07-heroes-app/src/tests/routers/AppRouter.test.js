import React from 'react';
import { mount } from 'enzyme';
import { AppRouter } from '../../routers/AppRouter';
import { AuthContext } from '../../auth/AuthContext';

describe('Pruebas en <AppRouter />', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    }

    test('should show login if not authenticate ', () => {
        
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect( wrapper ).toMatchSnapshot();

    })

    test('should show component marvel if is authenticate', () => {
        
        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'Peter'
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect( wrapper.find('.navbar').exists() ).toBe(true);
    })
    
    
})
