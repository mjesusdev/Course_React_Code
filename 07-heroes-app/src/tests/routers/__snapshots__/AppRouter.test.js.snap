// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> should show login if not authenticate  1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
